### PR TITLE
bokeh-server example of mock trading dashboard

### PR DESCRIPTION
**Pull request by caseyclements**
Same commit as had been requested to bokeh/bokeh/examples..
https://github.com/bokeh/bokeh/pull/2725

### CODE REVIEW

**Comment by caseyclements** [comment]
Please review

File: trading_dashboard/pnl_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    new_rate = TextInput(value="0.01", title="Fixed Rate:")
+    new_maturity = Select(title="Maturity:", value="1", options=['1','2','5','10','30'])
+    bookit = Button(label="Book it!", type="success")
+    bookit.on_click(bookit_clicked)
+
+    new_trade_entry = hplot(bookit, new_maturity, new_notional, new_rate)
+
+    # Create DataTable for New Trades
+    ds_new = ColumnDataSource(new_df)
+    print 'new_df:'
```
Comment (fpliger) [inline trading_dashboard/pnl_dashboard.py]: to remove before merging..

File: trading_dashboard/pnl_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    print new_df
+    #if ds_new is not None and len(ds_new.data) > 0:
+    new_trade_table = DataTable(source=ds_new, columns=columns, editable=True, width=1000)
+    return vplot(plot_rates, plot_pnl, data_table, new_trade_entry, new_trade_table)
+    #else:
```
Comment (fpliger) [inline trading_dashboard/pnl_dashboard.py]: better remove commented code before merging too

File: trading_dashboard/pnl_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    # table for today's new trades
+    new_df = init_df.drop(init_df.index) # same columns, no rows
+    dashboard = create(new_df, init_df)
+    show(dashboard)
+    # kick off 2-factor mean reverting process for yield curve
+    Thread(target=background_thread, args=(ds_prev,)).start()
+
+
+def background_thread_gaussian(ds):
+    """Plot animation, update data if play is True, otherwise stop"""
```
Comment (fpliger) [inline trading_dashboard/pnl_dashboard.py]: play is something that has been removed during development? If so, it's better to update the docstrings

File: trading_dashboard/pnl_dashboard.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    init_df = get_data()
+    # table for today's new trades
+    new_df = init_df.drop(init_df.index) # same columns, no rows
+    dashboard = create(new_df, init_df)
+    show(dashboard)
+    # kick off 2-factor mean reverting process for yield curve
+    Thread(target=background_thread, args=(ds_prev,)).start()
+
+
+def background_thread_gaussian(ds):
```
Comment (fpliger) [inline trading_dashboard/pnl_dashboard.py]: background_thread_gaussian doesn't seem to be used anywhere ATM. Is is something that we should keep? If so it'd be good to add a comment about how it is intended to be used and why it is not being used right now...

**Comment by fpliger** [comment]
Thanks for the addition @caseyclements ! I left some comments. Most are minor things related to print statements and commented code left there.. 

Another consideration is that it's not pep8 compliant and as there is need to change some things I'd suggest to the more compliant as possible.

The important comment is that it's python 2 only. We should try to support python 3 if this not a big issue.

Besides that it's works good locally on Python 2.7

**Comment by caseyclements** [comment]
Thanks @fpliger. I will fix up the cosmetic and vestigial bits, and make py3 compliant. 

I also will see if I can complete the original planned task - to add new trades that become part of the pnl calculation.

**Comment by caseyclements** [comment]
Hi - I am going to demo this dashboard at Strata. Would you please help to spruce this up?

**Comment by bryevdv** [comment]
@caseyclements @fpliger is already fairly overcommitted right now, can you be more specific about the kind of help yo need? Perhaps there is someone else that can help.

**Comment by bryevdv** [comment]
Closing, this repo will be deleted soon

**Pull request closed without merging**